## import Libraries

In [59]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

LOAD THE DATA

In [60]:
uci_bankData=pd.read_csv('bank-full-Interview.csv')

In [61]:
len(uci_bankData)

8417

Viewing the Data only Top few rows

In [62]:
uci_bankData.head()

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no   2143.0     yes   no   
1   44    technician   single  secondary      no     29.0     yes   no   
2   33  entrepreneur  married  secondary      no      2.0     yes  yes   
3   47   blue-collar  married    unknown      no   1506.0     yes   no   
4   33       unknown   single    unknown      no      1.0      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown  5.0   may     261.0       1.0   -1.0       0.0  unknown  no  
1  unknown  5.0   may     151.0       1.0   -1.0       0.0  unknown  no  
2  unknown  5.0   may      76.0       1.0   -1.0       0.0  unknown  no  
3  unknown  5.0   may      92.0       1.0   -1.0       0.0  unknown  no  
4  unknown  5.0   may     198.0       1.0   -1.0       0.0  unknown  no

Dataset Exploration
There are 8417 records with 17 fields in this Dataset
For this project we will consider y (customers who have term deposits) as the Dependent Field.
To predict whether customers will subscribe to Term Deposit , to check whether the bank's cross selling campaign will be justfied.

In [63]:
uci_bankData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8417 entries, 0 to 8416
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   age        8417 non-null   int64  
 1   job        8417 non-null   object 
 2   marital    8417 non-null   object 
 3   education  8417 non-null   object 
 4   default    8416 non-null   object 
 5   balance    8416 non-null   float64
 6   housing    8416 non-null   object 
 7   loan       8416 non-null   object 
 8   contact    8416 non-null   object 
 9   day        8416 non-null   float64
 10  month      8416 non-null   object 
 11  duration   8416 non-null   float64
 12  campaign   8416 non-null   float64
 13  pdays      8416 non-null   float64
 14  previous   8416 non-null   float64
 15  poutcome   8416 non-null   object 
 16  y          8416 non-null   object 
dtypes: float64(6), int64(1), object(10)
memory usage: 1.1+ MB


In [64]:
uci_bankData.describe()

age       balance          day     duration     campaign  \
count  8417.000000   8416.000000  8416.000000  8416.000000  8416.000000   
mean     39.373173   1022.954373    16.869653   264.025428     2.569986   
std       9.216843   2470.046375     8.697518   248.076378     2.884647   
min      20.000000  -3372.000000     2.000000     0.000000     1.000000   
25%      32.000000     39.000000     9.000000   117.000000     1.000000   
50%      38.000000    328.000000    16.000000   193.000000     2.000000   
75%      46.000000   1049.250000    26.000000   322.000000     3.000000   
max      61.000000  58544.000000    30.000000  3366.000000    63.000000   

        pdays  previous  
count  8416.0    8416.0  
mean     -1.0       0.0  
std       0.0       0.0  
min      -1.0       0.0  
25%      -1.0       0.0  
50%      -1.0       0.0  
75%      -1.0       0.0  
max      -1.0       0.0

In [ ]:
DATA CLEANING/Wrangling

In [ ]:
#renaming columns

In [65]:
uci_bankData.rename(columns={'duration':'duration of last contact','y':'Has Term Deposit?'})

age           job  marital  education default  balance housing loan  \
0      58    management  married   tertiary      no   2143.0     yes   no   
1      44    technician   single  secondary      no     29.0     yes   no   
2      33  entrepreneur  married  secondary      no      2.0     yes  yes   
3      47   blue-collar  married    unknown      no   1506.0     yes   no   
4      33       unknown   single    unknown      no      1.0      no   no   
...   ...           ...      ...        ...     ...      ...     ...  ...   
8412   29    technician  married    primary      no   1761.0     yes   no   
8413   30      services  married  secondary      no    227.0      no  yes   
8414   43    technician  married  secondary      no    463.0      no   no   
8415   54    management   single   tertiary      no      0.0     yes   no   
8416   41   blue-collar   single    seconda     NaN      NaN     NaN  NaN   

      contact  day month  duration of last contact  campaign  pdays  previous  \
0     unknown  5.0   may                     261.0       1.0   -1.0       0.0   
1     unknown  5.0   may                     151.0       1.0   -1.0       0.0   
2     unknown  5.0   may                      76.0       1.0   -1.0       0.0   
3     unknown  5.0   may                      92.0       1.0   -1.0       0.0   
4     unknown  5.0   may                     198.0       1.0   -1.0       0.0   
...       ...  ...   ...                       ...       ...    ...       ...   
8412  unknown  3.0   jun                     186.0       1.0   -1.0       0.0   
8413  unknown  3.0   jun                      23.0       1.0   -1.0       0.0   
8414  unknown  3.0   jun                     268.0       1.0   -1.0       0.0   
8415  unknown  3.0   jun                      68.0       1.0   -1.0       0.0   
8416      NaN  NaN   NaN                       NaN       NaN    NaN       NaN   

     poutcome Has Term Deposit?  
0     unknown                no  
1     unknown                no  
2     unknown                no  
3     unknown                no  
4     unknown                no  
...       ...               ...  
8412  unknown                no  
8413  unknown                no  
8414  unknown                no  
8415  unknown                no  
8416      NaN               NaN  

[8417 rows x 17 columns]

#How many customers have Term Deposit vs how many of them dont have a Term deposit

Retain only the columns required for Analysis
Convert non numerical but binary values (marital status, housing,loan

pd.get_dummies(uci_bankData['marital'])